src/psyclone/domain/lfric/lfric_builtins.py

```diff
@@ -557,6 +577,8 @@ def lower_to_language_level(self):
         rhs = BinaryOperation.create(BinaryOperation.Operator.ADD,
                                      arg_refs[1], arg_refs[2])
         assign = Assignment.create(arg_refs[0], rhs)
+        # Add a preceding comment to the Assignment
+        assign = LFRicBuiltIn.add_preceding_comment(self, assign)
```

Comment:
It's unfortunate (and not your fault) that the current design means we get this exact line duplicated in every `lower_to_language_level`. I see in fact that we already have the last three or so statements duplicated in every such method. Therefore, I think that instead of `add_preceding_comment` we have a 'private' `_create_assignment(lhs, rhs)` and then we can replace everything after (and including) `assign = Assignment.create(lhs, rhs)` with `return self._create_assignment(lhs, rhs)`. It's a slightly bigger change but should result in fewer lines of code overall.

Reply:
I was able to do this for all but one builtin - LFRicSetvalRandomKern. This is because there is no assignment node here, only IntrinsicCall.